File: src/main/java/co/gatedaccess/web/repo/MemberRepo.java
package co.gatedaccess.web.repo;

import co.gatedaccess.web.model.Member;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface MemberRepo extends MongoRepository<Member, String> {
    Member findMemberById(String userId);

    Member findMemberByInviteCode(String inviteCode);

    Member findMemberByEmail(String email);

    Boolean existsMemberByEmail(String email);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/MemberRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.Member
import org.springframework.data.mongodb.repository.MongoRepository

interface MemberRepo : MongoRepository<Member?, String?> {
    fun findMemberById(userId: String): Member?
    fun findByPhoneAndCommunityIsNotNull(phone: String): Member?
    fun findByPhone(phone: String): Member?
    fun existsByPhoneAndCommunityIsNotNull(phone: String): Boolean
}
